crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -84,6 +84,17 @@ impl<'w, E, B: Bundle> Trigger<'w, E, B> {
     }
 }
 
+impl<'w, E: Debug, B: Bundle> Debug for Trigger<'w, E, B> {
```

Comment:
```suggestion
/// Custom [`Debug`] implementation for [`Trigger`]
/// 
/// # Examples
/// 
/// ```rust
/// use bevy::prelude::*;
/// 
/// App::new()
///     .add_plugins(DefaultPlugins)
///     .add_systems(Update, click)
///     .observe(|trigger: Trigger<Example>| println!("{:#?}", trigger))
///     .run();
/// 
/// #[derive(Event, Debug)]
/// struct Example;
/// 
/// fn click(mouse_click: Res<ButtonInput<MouseButton>>, mut commands: Commands) {
///     if mouse_click.just_pressed(MouseButton::Left) {
///         commands.trigger(Example);
///     }
/// }
/// ```
/// 
/// will output something like
/// 
/// ```text
/// Trigger {
///     event: Example,
///     propagate: false,
///     trigger: ObserverTrigger {
///         observer: 1v1#4294967297,
///         event_type: ComponentId(
///             227,
///         ),
///         entity: 4294967295v1#8589934591,
///     },
///     _marker: PhantomData<()>,
/// }
/// ```
impl<'w, E: Debug, B: Bundle> Debug for Trigger<'w, E, B> {
```

Not sure how valuable the doc comment is but here's a quick example that also works as a test that triggers can actually be printed.

Reply:
That's a good idea, it could be even better to make the output an assertion so that it truly tests it. And personally I like the idea of keeping it on the `Debug` impl itself.

That said, imo I don't think we need to block on this